### DISCUSSION TITLE
save-demo

### DISCUSSION THREAD

**Discussion by sanyavertolet** [saveourtool/save-cloud#2144] [General]
As far as save-demo is more or less ready for public testing, I am to suggest finding big static analysis tools that do not have  online demo yet

**Comment by sanyavertolet** [comment]
## Python
https://github.com/pylint-dev/pylint
https://github.com/PyCQA/pydocstyle
https://github.com/PyCQA/flake8

**Comment by sanyavertolet** [comment]
## Kotlin
https://github.com/detekt/detekt
https://github.com/pinterest/ktlint

**Reply by nulls** [reply]
diktat

**Reply by sanyavertolet** [reply]
It already has demo

**Comment by sanyavertolet** [comment]
@akuleshov7 @nulls @0x6675636b796f75676974687562 any other suggestions?

**Reply by nulls** [reply]
maybe https://eslint.org/ or only jvm and python?

**Reply by sanyavertolet** [reply]
> maybe https://eslint.org/ or only jvm and python?

It depends on what we need to run js. Probably `ubuntu-latest` is not enough, so we might need a new docker image for save-demo-agent

**Comment by 0x6675636b796f75676974687562** [comment]
# Java

AFAIK, for Java, none of these tools has an online demo:

 - https://pmd.github.io
 - https://spotbugs.github.io
 - https://github.com/checkstyle/checkstyle

# C++

 - [Clang-Tidy](https://clang.llvm.org/extra/clang-tidy/) is the industry standard, yet I'm unaware of any demo available, either.
